### PR TITLE
Fix non-deterministic bug caused by unstableness in SSA 

### PR DESCRIPTION
**Pull request by sklam**
Fixes https://github.com/numba/numba/issues/9490

### CODE REVIEW

**Comment by sklam** [comment]
https://github.com/numba/numba/pull/9491/commits/d848d30f88d6fe804bffcfb7184e6c1fb928420f is needed to make the subprocess-test not accidentally running without isolation. What can happen is that the `TestParforsRunner` runner launches each `TestCase` class in subprocess with `SUBPROC_TEST=1` then any test that needs further isolation by using `@TestCase.run_test_in_subprocess` will pickup the `SUBPROC_TEST`. The change makes `run_test_in_subprocess` set and expects in the runner that `SUBPROC_TEST` is equal to the test name. The `TestParforsRunner` will still just set `SUBPROC_TEST=1`.

(CC @stuartarchibald)

**Comment by sklam** [comment]
`numba_smoketest_cpu_yaml_190`

**Comment by sklam** [comment]
Reviewer notes from triage meeting:

- accidental stableness by sort?
- will removal of assignments impact liveness hack?

**Comment by sklam** [comment]
> Reviewer notes from triage meeting:
> 
> * accidental stableness by sort?
> * will removal of assignments impact liveness hack?

https://github.com/numba/numba/pull/9491/commits/6b1de4e01bb708be47891d06bf48df2f2a160932 resolved these.

- `violators` is now based on insertion order which is in turn based on ordering of occurrence in bytecode.
- removal of self-assignments is not needed now. maybe i got it wrong initially because of non-det tests or that the use of insertion-order (above) helped.

**Comment by sklam** [comment]
`numba_smoketest_cpu_yaml_191`

**Comment by loicdtx** [comment]
In case that helps I applied some minor changes to the test case. See https://github.com/sklam/numba/pull/5

**Comment by sklam** [comment]
@loicdtx, Does the changes affect the ability to reproduce the error? or does it reveal a different problem? I found the test very sensitive to changes. If the changes are purely for algorithm correctness in `stable_fit`, I'd prefer to not apply them so that we preserve the failing characteristic.

**Comment by sklam** [comment]
I confirmed https://github.com/numba/numba/pull/9491/commits/0538805d812536f8d496576e636dcc7e902ad282 produces the same error with `PYTHONHASHSEED=1` without the fix.

File: numba/core/utils.py [hunk 1]
```diff
@@ -396,7 +418,7 @@ def __len__(self):
         return len(self._values)
 
     def __iter__(self):
-        return iter(k for k in sorted(self._values))
+        return iter(k for k in (self._values))
```
Comment (sklam) [inline numba/core/utils.py:421]:
```suggestion
        return iter(k for k in sorted(self._values))
```
Shouldn't have removed the `sorted`. It was part of debugging and i  accidentally added to the patch. It turns out only the RVSDG frontend uses this so no tests were affected.

**Review by guilhermeleobas — APPROVED** [review]
Can confirm the proposed patch fixes the bug (for now? 😅).

It would be good to know one day why the ordering matters here.

**Pull request merged**